#file is for custom exception classes
class SaveWriteException(Exception):
    """Exception for when writing to the save file fails"""

    def __init__(self, message, error_code):
        super().__init__(message)
        self.error_code = error_code

    def __str__(self):
        return f"{self.message} (Error Code: {self.error_code})"

class GatherDataException(Exception):
    """Exception for when gathering the data for saving fails"""

    def __init__(self, message, error_code):
        super().__init__(message)
        self.error_code = error_code

    def __str__(self):
        return f"{self.message} (Error Code: {self.error_code})"

class StartupGatherException(Exception):
    """Exception for when gathering data for startup fails"""

    def __init__(self, message, error_code):
        super().__init__(message)
        self.error_code = error_code

    def __str__(self):
        return f"{self.message} (Error Code: {self.error_code})"